core/lib/dr_config.h

```diff
@@ -153,6 +158,71 @@ typedef enum {
     DR_PLATFORM_NONE,    /**< Invalid platform. */
 } dr_platform_t;
 
+/**
+ * Information about a client library setup.
+ */
+typedef struct _dr_config_client_t {
+    /**
+     * The size of this structure.  This field must be set when passing as an input
+     * to functions like dr_get_client_info_ex().  Used for backward compatibility.
+     */
+    size_t struct_size;
+    /**
+     * This is a client_id_t uniquely identifying the client.  DynamoRIO provides the
+     * client ID as a parameter to dr_client_main().  Clients use this ID to retrieve
+     * client-specific path and option information. Outside entities also identify
+     * the target of a nudge via this ID.
+     */
+    client_id_t id;
+    /**
+     * The client number, or priority. Client registration includes a value
+     * indicating the priority of a client relative to other clients. In
+     * multi-client settings, a client's priority influences event callback
+     * ordering.That is, higher priority clients can register their callbacks first;
+     * DynamoRIO then calls these routines last.  Client priorities range
+     * consecutively from 0 to N-1, where N is the number of registered clients.
+     * Specify priority 0 to register a client with highest priority.
+     */
+    size_t priority;
+    /**
+     * A NULL-terminated string specifying the full path to a valid client library
+     * whose bitwidth is the opposite of that specified by dr_platform.  The string
+     * length cannot exceed #MAXIMUM_PATH.  The client path may not include any semicolons
+     * and when combined with \p client_options may not include all three quote
+     * characters (', ", `) simultaneously.
+     *
+     * When querying via dr_get_client_info_ex() or dr_client_iterator_next_ex(), the
+     * caller must either set this to NULL if that data is not needed or point at a
+     * caller-allocated array of length #MAXIMUM_PATH.
+     */
+    char *path;
+    /**
+     * A NULL-terminated string specifying options that are available to the client
+     * as arguments of dr_client_main() or via dr_get_option_array().  The string
+     * length cannot exceed #DR_MAX_OPTIONS_LENGTH.  The client options may not
+     * include any semicolons and when combined with \p client_path may not include
+     * all three quote characters (', ", `) simultaneously.
+     *
+     * When querying via dr_get_client_info_ex() or dr_client_iterator_next_ex(), the
+     * caller must either set this to NULL if that data is not needed or point at a
+     * caller-allocated array of length #DR_MAX_OPTIONS_LENGTH.
+     */
+    char *options;
+    /**
+     * Specifies whether this is a regular client registeration for the target
```

Comment:
I guess you don't *have* to: the client instance in a diff-bitwidth child would just get a different ID.  I reworded the unregister to say that it will unreg both if they have the same ID.